### PR TITLE
feat(js): lightbox href calls now uses elgg/Ajax

### LINKED ISSUE #10078: Teach lightbox to use elgg/Ajax
Anything loaded in a lightbox can not be filtered via Ajax service hooks, required modules are not loaded etc

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #10078

### CODE REVIEW

**Comment by jdalsem** [comment]
can't use onLoad callbacks or cbox_load event as they can not be asynchronous

File: views/default/elgg/lightbox.js [hunk 1]
```diff
@@ -107,7 +105,29 @@ define(function (require) {
 						opts.href = elgg.getSelectorFromUrlFragment(opts.href);
 					}
 
-					lightbox.open(opts);
+					if (opts.photo || opts.inline || opts.iframe || opts.html) {
+						lightbox.open(opts);
+					} else {
+						
+						var href = opts.href
```
Comment (mrclay) [inline views/default/elgg/lightbox.js]: Needs semicolon

File: views/default/elgg/lightbox.js [hunk 2]
```diff
@@ -107,7 +105,29 @@ define(function (require) {
 						opts.href = elgg.getSelectorFromUrlFragment(opts.href);
 					}
 
-					lightbox.open(opts);
+					if (opts.photo || opts.inline || opts.iframe || opts.html) {
+						lightbox.open(opts);
```
Comment (mrclay) [inline views/default/elgg/lightbox.js]: Just return after this?

File: views/default/elgg/lightbox.js [hunk 3]
```diff
@@ -107,7 +105,29 @@ define(function (require) {
...
+						
+						require(['elgg/Ajax'], function(Ajax) {
+							var ajax = new Ajax(false);
+							ajax.path(href, {data: data}).done(function(output) {
+								opts.html = output;
+								lightbox.open(opts);
+								
+								// clear data so next fetch will refresh contents
+								opts.html = undefined;
```
Comment (mrclay) [inline views/default/elgg/lightbox.js]: Are you certain this is necessary?

**Review by mrclay — CHANGES_REQUESTED** [review]

**Comment by mrclay** [comment]
I guess a downside here is unexpected colorbox events.

File: views/default/elgg/lightbox.js (hunk 3 above)
Comment (jdalsem) [inline views/default/elgg/lightbox.js]: 100%... if you reopen the lightbox... the content would otherwise still be there (and you would see no loader)

**Comment by jdalsem** [comment]
> I guess a downside here is unexpected colorbox events.

like? because of the extra 'open'?

File: views/default/elgg/lightbox.js (hunk 2 above)
Comment (jdalsem) [inline views/default/elgg/lightbox.js]: done

File: views/default/elgg/lightbox.js [hunk 4]
```diff
@@ -107,7 +105,29 @@ define(function (require) {
...
 					lightbox.open(opts);
+					
+					require(['elgg/Ajax'], function(Ajax) {
+						var ajax = new Ajax(false);
+						ajax.path(href, {data: data}).done(function(output) {
+							opts.html = output;
+							lightbox.open(opts);
+							
+							// clear data so next fetch will refresh contents
+							opts.html = undefined;
```
Comment (hypeJunction) [inline views/default/elgg/lightbox.js]: Let's trigger jQuery events here, similar to popup and tabs

Comment (jdalsem) [inline views/default/elgg/lightbox.js]: why? we have events in colorbox....

Comment (hypeJunction) [inline views/default/elgg/lightbox.js]: It's easier to listen to jQuery events.

**Comment by jdalsem** [comment]
If we want extra 'special' jquery events we can always add those later. It is not needed for this PR, so merging this.

**Pull request merged**